flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherGateway.java

```diff
@@ -82,4 +82,8 @@
 	default CompletableFuture<Acknowledge> shutDownCluster(ApplicationStatus applicationStatus) {
 		return shutDownCluster();
 	}
+
+	default CompletableFuture<Acknowledge> shutDownClusterExceptionally(final Throwable throwable) {
+		throw new UnsupportedOperationException();
+	}
```

Comment:
I think we should not introduce this method. Instead the `ApplicationDispatcherBootstrap` should get a `FatalErrorHandler` which it simply calls if it sees an unknown exception.